### ISSUE TITLE
bug: ClickHouse implementation missing method UseTracer

### ISSUE THREAD

**Issue by mfreeman451** [gofr-dev/gofr#1068]
**Describe the bug**
I tried to run the example code in the documentation and am getting errors:

**To Reproduce**
Steps to reproduce the behavior, if applicable:

1. The code is

Taken from here

https://gofr.dev/docs/advanced-guide/injecting-databases-drivers

```go
   package main

import (
    "gofr.dev/pkg/gofr"

    "gofr.dev/pkg/gofr/datasource/clickhouse"
)

type User struct {
    Id   string `ch:"id"`
    Name string `ch:"name"`
    Age  string `ch:"age"`
}

func main() {
    app := gofr.New()

    app.AddClickhouse(clickhouse.New(clickhouse.Config{
        Hosts:    "localhost:9001",
        Username: "root",
        Password: "password",
        Database: "users",
    }))
    
    app.POST("/user", Post)
    app.GET("/user", Get)
    
    app.Run()
}

func Post(ctx *gofr.Context) (interface{}, error) {
    err := ctx.Clickhouse.Exec(ctx, "INSERT INTO users (id, name, age) VALUES (?, ?, ?)", "8f165e2d-feef-416c-95f6-913ce3172e15", "aryan", "10")
    if err != nil {
        return nil, err
    }

    return "successful inserted", nil
}

func Get(ctx *gofr.Context) (interface{}, error) {
    var user []User

    err := ctx.Clickhouse.Select(ctx, &user, "SELECT * FROM users")
    if err != nil {
        return nil, err
    }

    return user, nil
}
```

2. The error is

   ```
   ./main.go:18:23: cannot use clickhouse.New(clickhouse.Config{…}) (value of type *"gofr.dev/pkg/gofr/datasource/clickhouse".client) as container.ClickhouseProvider value in argument to app.AddClickhouse: *"gofr.dev/pkg/gofr/datasource/clickhouse".client does not implement container.ClickhouseProvider (missing method UseTracer)
   ```

**Expected behavior**
No errors

**Screenshots**

**Environments (please complete the following information):**
- OS: OSX
- gofr version 
```
        gofr.dev v1.22.0
        gofr.dev/pkg/gofr/datasource/clickhouse v0.0.0-20240813061059-ef39e204e25a
```
- go version 1.23.1 darwin/arm64

**More description**

**Comment by aryanmehrotra** [comment]
It has been fixed in v1.22.1

**Comment by mfreeman451** [comment]
I don't think it was fixed: ```go: upgraded gofr.dev v1.22.0 => v1.22.1
go: upgraded golang.org/x/exp v0.0.0-20231108232855-2478ac86f678 => v0.0.0-20240909161429-701f63a606c0
go: upgraded google.golang.org/genproto v0.0.0-20240903143218-8af14fe29dc1 => v0.0.0-20240930140551-af27646dc61f
go: upgraded google.golang.org/genproto/googleapis/api v0.0.0-20240903143218-8af14fe29dc1 => v0.0.0-20240930140551-af27646dc61f
go: upgraded google.golang.org/genproto/googleapis/rpc v0.0.0-20240903143218-8af14fe29dc1 => v0.0.0-20240930140551-af27646dc61f
go: upgraded google.golang.org/grpc v1.67.0 => v1.67.1
go: upgraded modernc.org/gc/v3 v3.0.0-20240107210532-573471604cb6 => v3.0.0-20240801135723-a856999a2e4a
go: upgraded modernc.org/libc v1.55.3 => v1.61.0
❯ go run .
# goo.com
./main.go:18:23: cannot use clickhouse.New(clickhouse.Config{…}) (value of type *"gofr.dev/pkg/gofr/datasource/clickhouse".client) as container.ClickhouseProvider value in argument to app.AddClickhouse: *"gofr.dev/pkg/gofr/datasource/clickhouse".client does not implement container.ClickhouseProvider (missing method UseTracer)```
